packages/abstractions/kiota_abstractions/date_utils.py

```diff
@@ -0,0 +1,76 @@
+import re
+from datetime import timedelta
+
+
+_ISO8601_DURATION_PATTERN = re.compile(
+    "^P"  # Duration P indicator
+    # Weeks
+    "(?P<w>"
+    r"    (?P<weeks>\d+(?:[.,]\d+)?W)"
```

Comment:
are those `?P` a special syntax in Python regex for group naming? or are we looking for the mandatory P character in all sub-segments?

Reply:
Yes ?P is python regex for, this is a named group